packages/kbn-unified-data-table/src/components/data_table.scss

```diff
@@ -161,3 +161,12 @@
     display: none;
   }
 }
+
+// Ensure full screen data grids are not covered by elements with a z-index
+.euiDataGrid__restrictBody *:not(
+.euiDataGrid--fullScreen,
+.euiDataGrid--fullScreen *,
+[data-euiportal='true'],
+[data-euiportal='true'] *) {
+  z-index: unset !important;
+}
```

Comment:
These styles are (necessarily) global for consumers of Unified Data Table. I couldn't figure out another way to fix the issue resulting from `EuiDataGrid`'s usage of `fixed` positioning without a significant change to how it functions in EUI (e.g. using a portal and moving it in the DOM instead of using CSS). I'm open to any thoughts or suggestions here though.

Alternatively it may be worth making these styles global throughout Kibana by adding them to one of the core style sheets instead. Figured I'd implement at the Unified Data Table level first and see what others think.

Reply:
This seems like affecting many DOM elements. What if we address only the mentioned issue and hide the code editor for example?:

```
.euiDataGrid__restrictBody <add a discover class name> .uniSearchBar {
  visibility: hidden;
}
```

Reply:
Yes, it's intentionally affecting all DOM elements outside of the data grid. It's not an optimal solution, but I'm not sure of another approach that works reliably without extensive changes to the component in EUI. The problem with targeting Unified Search directly is that the issue occurs with other elements too, like the Unified Histogram resize handle (I used that one for the functional test).